Assets/MixedRealityToolkit/Inspectors/Setup/BuildDeployWindow.cs

```diff
@@ -0,0 +1,1436 @@
+                foreach (string appBuild in buildList)
+                {
+                    if (appBuild.Contains(appxDirectory) && !appBuild.Contains($"{appxDirectory}\\"))
+                    {
+                        AppPackageDirectories.AddRange(Directory.GetDirectories(appBuild));
+                    }
+                }
+
+                IEnumerable<string> selectedDirectories =
+                    from string directory in AppPackageDirectories
+                    orderby Directory.GetLastWriteTime(directory) descending
+                    select Path.GetFullPath(directory);
+                Builds.AddRange(selectedDirectories);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                // unused
+            }
+
+            UpdatePackageName();
+        }
+
+        private static string CalcMostRecentBuild()
+        {
+            UpdateBuilds();
+            DateTime mostRecent = DateTime.MinValue;
+            string mostRecentBuild = string.Empty;
+
+            foreach (var fullBuildLocation in Builds)
+            {
+                DateTime directoryDate = Directory.GetLastWriteTime(fullBuildLocation);
+
+                if (directoryDate > mostRecent)
+                {
+                    mostRecentBuild = fullBuildLocation;
+                    mostRecent = directoryDate;
+                }
+            }
+
+            return mostRecentBuild;
+        }
+
+        private void UpdatePortalConnections()
+        {
+            targetIps = new string[portalConnections.Connections.Count];
+
+            for (int i = 0; i < targetIps.Length; i++)
+            {
+                var connection = portalConnections.Connections[i];
+                string machineName = connection.MachineName;
+                if (string.IsNullOrEmpty(machineName) && IsLocalConnection(connection))
+                {
+                    connection.MachineName = IsHoloLensConnectedUsb ? HOLOLENS_USB : LOCAL_MACHINE;
+                }
+
+                targetIps[i] = machineName + " - " + connection.IP;
+            }
+
+            UwpBuildDeployPreferences.DevicePortalConnections = JsonUtility.ToJson(portalConnections);
+            lastSessionConnectionInfoIndex = CurrentConnectionIndex;
+            Repaint();
+        }
+
+        private static bool IsLocalConnection(DeviceInfo connection)
+        {
+            return connection.IP.Equals(LOCAL_MACHINE, StringComparison.OrdinalIgnoreCase) ||
+                   connection.IP.Equals(LOCAL_IP_ADDRESS);
+        }
+
+        private static bool AreCredentialsValid(DeviceInfo connection)
+        {
+            return !string.IsNullOrEmpty(connection.User) &&
+                   !string.IsNullOrEmpty(connection.IP) &&
+                   !string.IsNullOrEmpty(connection.Password);
+        }
+
+        private static bool IsIPAddress(string ip)
+        {
+            return IPAddress.TryParse(ip, out IPAddress address);
```

Comment:
Also, I think this might miss some other cases the previous version accounted for. From the `IPAddress.TryParse` docs:

>[...] if ipString is "1", this method returns true even though "1" (or 0.0.0.1) is not a valid IP address and you might expect this method to return false. [...] Your code can avoid this behavior by ensuring that it only uses this method to parse IP addresses in dotted-decimal format

Previously there was a split on `.` to ensure the address was formatted correctly:

```c#
var subAddresses = ip.Split('.');	
return subAddresses.Length > 3;
```